client/web/src/site-admin/notifications/NotificationsNavbarItem.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useMemo } from 'react'
+
+import { mdiBellAlert, mdiBellBadge, mdiBellCheck, mdiCheckCircle, mdiInformation } from '@mdi/js'
+import classNames from 'classnames'
+import { groupBy, startCase } from 'lodash'
+
+import { renderMarkdown } from '@sourcegraph/common'
+import { Icon, Badge, Popover, PopoverTrigger, PopoverContent, Text, Button, PopoverTail } from '@sourcegraph/wildcard'
+
+import { AlertType } from '../../graphql-operations'
+
+import { useNotifications } from './hooks/useNotifications'
+
+import styles from './NotificationsNavbarItem.module.scss'
+
+const NotificationsGroup: React.FC<{ name: string; items: { message: string; type: AlertType }[] }> = ({
+    name,
+    items,
+}) => {
+    const hasError = items.find(item => item.type === AlertType.ERROR)
+    const hasWarning = items.find(item => item.type === AlertType.WARNING)
+    const hasInfo = items.find(item => item.type === AlertType.INFO)
```

Comment:
How large do you think this `items` array can be? It may be efficient to loop through this array once instead of thrice.

```suggestion
	let hasError: boolean = false
	let hasWarning: boolean = false
	let hasInfo: boolean = false
	
	items.forEach(item => {
		switch (item.type) {
			case AlertType.Error:
				hasError = true
				break
			case AlertType.WARNING:
				hasWarning = true
				break
			case AlertType.INFO:
				hasInfo = true
				break
		}
	})
```

Reply:
Thanks for raising. An array is not large less than 10 for sure, so I would like to leave it as is if you don't mind, for the sake of readability.